### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by matt-paul**
- [x] Allow website logging in to website via github, using auth0, in readiness for 'logged-in ' features such as the events page.
 - [x] Fix all linting errors across project
 - [x] Add Eslint to circleci
 - [x] Add Flow to circleci

Because fixing lint errors has touched a lot of code, a thorough test is suggested, especially on Stripe and pipedrive related functionality.  In some area's where a little more thought was needed to fixing lint errors, I have disabled rule or ignored the file.   It might be useful to fix these errors over time properly.

###Testing strategy:
The login button is currently hidden behind a simple feature flag.  To view the login button (and logout button when logged in), add url parameters with 'login' feature
ie http://localhost:3000/?ft=login
Once logged, you will see the 'events' page within the navigation bar.

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,52 +0,0 @@
-.App {
```
Comment (felixmeziere) [inline src/App.css]: 👍

File: .eslintrc [hunk 2]
```diff
@@ -1,32 +1,31 @@
 {
-  "extends": "airbnb",
+  "extends": ["airbnb", "prettier"],
```
[the remark this answers is not in the document]
Comment (matt-paul) [reply .eslintrc]: agreed on this order with Felix

File: lambda/functions/stripe.js [hunk 3]
```diff
@@ -8,7 +9,7 @@ export default (event, context, callback) => {
 
     getSSMParameter(process.env.STRIPE_PRIVATE_KEY_ALIAS)
       .then(stripePrivateKey => stripeLoader(stripePrivateKey))
-      .then(stripe => (
+      .then(stripe =>
```
[the remark this answers is not in the document]
Comment (matt-paul) [reply lambda/functions/stripe.js]: as discussed

File: lambda/functions/pipedrive/actions/addStudent.js [hunk 4]
```diff
@@ -28,13 +28,11 @@ const addStudent = (data, pipedrive, callback) => {
 
   return pipedrive.Persons.add(addPersonRequestBody, (error, successData) => {
     if (error) {
-      console.error(error);
```
[the remark this answers is not in the document]
Comment (matt-paul) [reply lambda/functions/pipedrive/actions/addStudent.js]: fixed!

File: lambda/functions/pipedrive/actions/addStudent.js [hunk 5]
```diff
@@ -28,13 +28,11 @@ const addStudent = (data, pipedrive, callback) => {
...
       return callback(new Error('[400] Error while adding student.'));
     }
     addDealRequestBody.person_id = successData.id;
-    return pipedrive.Deals.add(addDealRequestBody, (error2) => {
+    return pipedrive.Deals.add(addDealRequestBody, error2 => {
       if (error2) {
-        console.error(error2);
```
[the remark this answers is not in the document]
Comment (matt-paul) [reply lambda/functions/pipedrive/actions/addStudent.js]: fixed!

File: lambda/functions/stripe.js [hunk 6]
```diff
@@ -17,14 +18,8 @@ export default (event, context, callback) => {
             source: token,
           })
           .then(() => callback(null, '[200] Payment succesful.'))
-          .catch((error) => {
-            console.error(error);
```
[the remark this answers is not in the document]
Comment (matt-paul) [reply lambda/functions/stripe.js]: fixed!

File: lambda/functions/stripe.js [hunk 7]
```diff
@@ -17,14 +18,8 @@ export default (event, context, callback) => {
...
-            return callback(error, '[400] Error during payment.');
-          })
-      ))
-      .catch((error) => {
-        console.error(error);
```
[the remark this answers is not in the document]
Comment (matt-paul) [reply lambda/functions/stripe.js]: fixed!

File: src/components/navigation/index.js [hunk 8]
```diff
@@ -64,30 +75,47 @@ const Navigation = () => (
...
         </NavItem>
-        <NavItem className="nav-icon" eventKey={'linkedin'}>
+        <NavItem className="nav-icon" eventKey="linkedin">
           <img src={linkedinlogo} style={logoStyle} alt="linkedin" />
         </NavItem>
-        <NavItem className="nav-icon" eventKey={'email'}>
+        <NavItem className="nav-icon" eventKey="email">
           <img src={emaillogo} style={logoStyle} alt="email" />
         </NavItem>
+        {setFeatures().active('login') && (
```
Comment (kkarimi) [inline src/components/navigation/index.js]: Not clear on how this works

File: src/components/navigation/index.js [hunk 9]
```diff
@@ -45,7 +52,11 @@ const handleLink = linkName => {
   window.open(URL, '_blank');
 };
 
-const Navigation = () => (
+type Props = {
+  auth: Object,
```
Comment (kkarimi) [inline src/components/navigation/index.js]: should this be same `auth` type as in `./src/pages/profile/index.js`?

File: .prettierrc [hunk 10]
```diff
@@ -1,6 +1,7 @@
 {
-	"bracketSpacing": true,
-	"singleQuote": true,
-	"tabWidth": 2,
-	"useTabs": false
+  "bracketSpacing": true,
+  "singleQuote": true,
+  "tabWidth": 2,
+  "useTabs": false,
```
Comment (kkarimi) [inline .prettierrc]: :smirk:

File: circle.yml [hunk 11]
```diff
@@ -19,9 +19,14 @@ jobs:
...
-          command: |
-            yarn test
+          name: Lint project code
+          command: yarn lint:js
+      - run:
+          name: Run unit tests
+          command: yarn test
+      - run:
+          name: Run Flow
+          command: yarn flow
```
Comment (kkarimi) [inline circle.yml]: :heart_eyes:

File: lambda/functions/stripe.js [hunk 12]
```diff
@@ -17,12 +18,14 @@ export default (event, context, callback) => {
             source: token,
           })
           .then(() => callback(null, '[200] Payment succesful.'))
-          .catch((error) => {
+          .catch(error => {
+            // eslint-disable-next-line no-console
```
Comment (kkarimi) [inline lambda/functions/stripe.js]: Mhm yes maybe we should consider/add a ticket for `sentry.io` integration

File: src/pages/create-event/index.js [hunk 13]
```diff
@@ -0,0 +1,116 @@
+// @flow
```
Comment (kkarimi) [inline src/pages/create-event/index.js]: Guess this is just a placeholder?

File: lambda/functions/stripe.js (hunk 12 above)
Comment (matt-paul) [reply lambda/functions/stripe.js]: 👍

File: src/components/navigation/index.js (hunk 8 above)
Comment (matt-paul) [reply src/components/navigation/index.js]: @kkarimi using https://github.com/ericelliott/feature-toggle
There is probalby a better way to do this going forward, but this seemed to do the trick for now

File: src/pages/create-event/index.js (hunk 13 above)
Comment (matt-paul) [reply src/pages/create-event/index.js]: That is left in from the previous attempt at doing this, I left it in as I thought there might be a couple of things i could make use of to get started

File: src/components/navigation/index.js (hunk 9 above)
Comment (matt-paul) [reply src/components/navigation/index.js]: good spot.. I have moved the Auth type to a types directory and referenced it here.  Most types i have added at this point are pretty basic but over time we can improve them

File: src/components/navigation/index.js (hunk 8 above)
Comment (kkarimi) [reply src/components/navigation/index.js]: +1

**Pull request merged**